### ISSUE TITLE
Nested data heading level accessibility failure

### ISSUE THREAD

**Issue by jackrhumphrey** [anvilco/spectaql#468]
Hi we've been making use of the data API in our SpectaQL implementation.

Ideally we would nest our data at 3 levels: Reference, Component and Mutation/Query/Type. This is supported in the example file:

// What should be noted is that you can nest things (more than once) for ultimate
// control over what data you display, and in what arrangement. Yay!

However we found an accessibility failure when doing so:

![specta](https://user-images.githubusercontent.com/22863763/186039211-5b7bd72a-cb7b-44e6-9ae9-dcf66c5c2079.png)

The 3rd level heading is an h4 and not an h6 as you would expect and it is also not clickable

As a workaround we have bumped our headings down a level, so the top level headings are "Core"/"User/"Web API" etc. Ideally we would nest those inside of Reference as the top level heading however

Heading levels stop at h6 but additional levels could be defined using aria-level https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA12 we don't need headings beyond h6 though.

Thanks

**Comment by newhouse** [comment]
@jackrhumphrey 

I'm working on this for release with v2.

What I have built now will do the following:
In the nav panel, the heading will be `<h1>` through `<h6>` as depth increases.
But at depths greater than `6`, it will become `<h{depth} ... aria-level={depth}>`. 
So, for example, at a depth of `8`, it will be: `<h8 aria-level="8">`


Note: you will only see `aria-level` when things get deeper than `6`.

How does that feel/seem?
